src/random-seed/random-seed.c

```diff
@@ -242,6 +245,11 @@ static int run(int argc, char *argv[]) {
                         if (r < 0)
                                 log_error_errno(r, "Failed to write seed to /dev/urandom: %m");
                 }
+                if (write_seed_file) {
+                        sha256_init_ctx(&hash_state);
+                        sha256_process_bytes(&buf_size, sizeof(buf_size), &hash_state);
+                        sha256_process_bytes(buf, buf_size, &hash_state);
```

Comment:
shouldn't these two lines use `k` (read result) instead of `buf_size` (allocated size)?

Reply:
Yes indeed, thanks.